### PR TITLE
INT-6741 Create support script to get plugins details

### PR DESCRIPTION
**Pull request by hectorhuol**
Adding a script to get information about the plugins installed on the Jenkins server. It gets information about the plugins, the loaded packages by each plugin, and how Jenkins is loading some classes.

As we haven't been able to reproduce the issue reported on the JIRA ticket, we are creating this script to try to get a picture of the customer's Jenkins server to try to reproduce the issue locally.

JIRA: https://issues.sonatype.org/browse/INT-6741
Build: https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/feature-snapshots/job/INT-6741-support-script-to-get-plugin-details/

### CODE REVIEW

**Comment by jflinchbaugh** [comment]
would you mind adding a quick readme.md to the support directory to explain these files are useful support/diagnostic scripts or whatever and how they're to be run (probably jenkins groovy console)?

**Review by eduard-tita — APPROVED** [review]
+1 with one suggestion

File: support/print-plugins-information-script.groovy [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/**
```
Comment (jflinchbaugh) [inline support/print-plugins-information-script.groovy:1]: will the license check tool complain about this file for not having the copyright/license header? ⌛

File: support/print-plugins-information-script.groovy [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    println "Found: ${plugin}"
+    println '-------------------------------------------------------------'
+    println "BackupVersion: ${plugin.getBackupVersion()}"
+    println "Dependents: ${plugin.getDependents()}"
+    println "Dependencies: ${plugin.getDependencies()}"
+    println "RequiredCoreVersion: ${plugin.getRequiredCoreVersion()}"
+    println "Version: ${plugin.getVersion()}"
+    println "VersionNumber: ${plugin.getVersionNumber()}"
+    println "DerivedDependencyErrors: ${plugin.getDerivedDependencyErrors()}"
+    println ''
```
Comment (jflinchbaugh) [inline support/print-plugins-information-script.groovy:68]: i'd see if you can use multiline strings and stripIndent() in the jenkins console:
println """
    Found: ${plugin}
    ---...
    BackupVersion: ...
""".stripIndent()

File: support/print-plugins-information-script.groovy [hunk 3]
```diff
@@ -0,0 +1,88 @@
+/**
+ * This script is intended for support purposes only. The script do not get any sensitive information, it only gets
+ * information about the installed plugins on the Jenkins server, the packages loaded by each plugin, and information
+ * about how some classes are loaded by Jenkins.
+ *
+ * To run the script, you will need to go to the Script Console on the Jenkins server, paste the script, and hit the
+ * "Run" button. Here you have more information:
+ * ---> https://www.jenkins.io/doc/book/managing/script-console/
```
Comment (jflinchbaugh) [inline support/print-plugins-information-script.groovy:8]: 👍

File: support/print-plugins-information-script.groovy (hunk 1 above)
Comment (hectorhuol) [reply support/print-plugins-information-script.groovy:1]: Good point, I am going to add the header

Comment (hectorhuol) [reply support/print-plugins-information-script.groovy:14]: Done

File: support/print-plugins-information-script.groovy [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ * Prints information about how Jenkins is loading some classes
+ */
+def printPluginsLoadingClasses() {
+  println 'Looking for Plugins Loading Classes'
+  println '############################################################'
+  println ''
+  def classesToFind = [
+      "com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl",
+      "org.cyclonedx.generators.xml.AbstractBomXmlGenerator",
```
[the remark this answers is not in the document]
Comment (hectorhuol) [reply support/print-plugins-information-script.groovy]: Done

File: support/print-plugins-information-script.groovy [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  println "Class found in: ${list}"
+  println ''
+}
+
+/**
+ * Prints information about installed plugins on Jenkins server
+ */
+def printPluginsInformation() {
+  println 'Looking for Plugins Information'
+  println '############################################################'
```
[the remark this answers is not in the document]
Comment (hectorhuol) [reply support/print-plugins-information-script.groovy]: Oh, this is new for me, it is fun 😄. Done.

File: support/print-plugins-information-script.groovy (hunk 2 above)
Comment (hectorhuol) [reply support/print-plugins-information-script.groovy]: Yeah, I can do that. Done

**Comment by hectorhuol** [comment]
> would you mind adding a quick readme.md to the support directory to explain these files are useful support/diagnostic scripts or whatever and how they're to be run (probably jenkins groovy console)?

Sounds good, I just added the `README` file

File: support/print-plugins-information-script.groovy (hunk 4 above)
[the remark this answers is not in the document]
Comment (hectorhuol) [reply support/print-plugins-information-script.groovy]: Done

**Review by jflinchbaugh — APPROVED** [review]
+1

**Pull request merged**